yt/data_objects/construction_data_containers.py

```diff
@@ -638,7 +639,10 @@ def __init__(
         self.ActiveDimensions = self._sanitize_dims(dims)
 
         rdx = self.ds.domain_dimensions * self.ds.relative_refinement(0, level)
-        rdx[np.where(np.array(dims) - 2 * num_ghost_zones <= 1)] = 1  # issue 602
+
+        # normalize dims as a non-zero dim array
+        dims = np.array(list(always_iterable(dims)))
+        rdx[np.where(dims - 2 * num_ghost_zones <= 1)] = 1  # issue 602
```

Comment:
What is the purpose of `np.where` here? Can't we just use 
```suggestion
        rdx[dims - 2 * num_ghost_zones <= 1] = 1  # issue 602
```

Reply:
Looks reasonable to me, though I can't test it right now because I'm afk, and I'm never confident that I understand no.where. Give me a screenshot and I'll commit this :)

Reply:
![image](https://user-images.githubusercontent.com/5411875/123987570-96ffd500-d9c7-11eb-9132-e993a2b800d0.png)

Reply:
It is faster, but this only gets called once for each grid, and the rdx sizes are quite small.

Reply:
still, it's also more readable, so win+win. I was only afraid it might not be equivalent. Committing !

Reply:
Here's the problem:

```
> >       rdx[dims - 2 * num_ghost_zones <= 1] = 1  # issue 602
910
E       IndexError: boolean index did not match indexed array along dimension 0; dimension is 3 but corresponding boolean dimension is 1
```

this is from the minimal build CI. Looks like on old versions of numpy, your suggestion doesn't work. Either that or we overlooked something. In any case, I'll revert.